From e8f7edd463e294a2aa84b154f6f3c59bfde6f4d4 Mon Sep 17 00:00:00 2001
From: Grygorii Strashko <grygorii.strashko@ti.com>
Date: Sat, 12 Jun 2021 04:27:57 +0300
Subject: [PATCH] HACK: net: ethtool: export convert
 legacy_settings_to_link_ksettings func

Export convert_legacy_settings_to_link_ksettings() and
convert_link_ksettings_to_legacy_settings() which are required to process
data for user space commands CONFIG_SWITCH_GET/SET_PORT_CONFIG.

Signed-off-by: Grygorii Strashko <grygorii.strashko@ti.com>
Signed-off-by: Vignesh Raghavendra <vigneshr@ti.com>
---
 include/linux/ethtool.h | 6 ++++++
 net/ethtool/common.c    | 1 +
 net/ethtool/ioctl.c     | 3 ++-
 3 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/include/linux/ethtool.h b/include/linux/ethtool.h
index 6408b446051f..9c710c1320d0 100644
--- a/include/linux/ethtool.h
+++ b/include/linux/ethtool.h
@@ -193,6 +193,12 @@ void ethtool_convert_legacy_u32_to_link_mode(unsigned long *dst,
 bool ethtool_convert_link_mode_to_legacy_u32(u32 *legacy_u32,
 				     const unsigned long *src);
 
+bool convert_legacy_settings_to_link_ksettings(struct ethtool_link_ksettings *link_ksettings,
+					       const struct ethtool_cmd *legacy_settings);
+
+bool convert_link_ksettings_to_legacy_settings(struct ethtool_cmd *legacy_settings,
+					       const struct ethtool_link_ksettings *link_ksettings);
+
 #define ETHTOOL_COALESCE_RX_USECS		BIT(0)
 #define ETHTOOL_COALESCE_RX_MAX_FRAMES		BIT(1)
 #define ETHTOOL_COALESCE_RX_USECS_IRQ		BIT(2)
diff --git a/net/ethtool/common.c b/net/ethtool/common.c
index 24036e3055a1..1bd7872eb091 100644
--- a/net/ethtool/common.c
+++ b/net/ethtool/common.c
@@ -330,6 +330,7 @@ convert_legacy_settings_to_link_ksettings(
 		= legacy_settings->eth_tp_mdix_ctrl;
 	return retval;
 }
+EXPORT_SYMBOL_GPL(convert_legacy_settings_to_link_ksettings);
 
 int __ethtool_get_link(struct net_device *dev)
 {
diff --git a/net/ethtool/ioctl.c b/net/ethtool/ioctl.c
index 2917af3f5ac1..e6d8d5212c5c 100644
--- a/net/ethtool/ioctl.c
+++ b/net/ethtool/ioctl.c
@@ -365,7 +365,7 @@ EXPORT_SYMBOL(ethtool_convert_link_mode_to_legacy_u32);
 /* return false if ksettings link modes had higher bits
  * set. legacy_settings always updated (best effort)
  */
-static bool
+bool
 convert_link_ksettings_to_legacy_settings(
 	struct ethtool_cmd *legacy_settings,
 	const struct ethtool_link_ksettings *link_ksettings)
@@ -407,6 +407,7 @@ convert_link_ksettings_to_legacy_settings(
 		= link_ksettings->base.transceiver;
 	return retval;
 }
+EXPORT_SYMBOL_GPL(convert_link_ksettings_to_legacy_settings);
 
 /* number of 32-bit words to store the user's link mode bitmaps */
 #define __ETHTOOL_LINK_MODE_MASK_NU32			\
-- 
2.32.0 (Apple Git-132)

